htlcswitch/link.go

```diff
@@ -2829,18 +2829,42 @@ func (l *channelLink) processRemoteSettleFails(fwdPkg *channeldb.FwdPkg,
 
 			l.log.Debugf("Failed to send %s", pd.Amount)
 
-			// If the failure message lacks an HMAC (but includes
-			// the 4 bytes for encoding the message and padding
-			// lengths, then this means that we received it as an
-			// UpdateFailMalformedHTLC. As a result, we'll signal
-			// that we need to convert this error within the switch
-			// to an actual error, by encrypting it as if we were
-			// the originating hop.
-			convertedErrorSize := lnwire.FailureMessageLength + 4
-			if len(pd.FailReason) == convertedErrorSize {
```

Comment:
Need to think about this deletion in the context of upgrade scenarios with lingering 'old style' malformed fails.

Reply:
Migrating `Fail` to `MalformedFail` for the pending updates in the channel db doesn't look very attractive.